robot/basestation/static/js/roslibjs/ros-helpers.js

```diff
@@ -1,6 +1,23 @@
 REQUEST_TIMEOUT = 3000
 ROTATE_TIMEOUT = 1000
+const PING_THROTTLE_TIME = 1000
 lastRotate = 0
+var lastCMDSent = 0
+
+// reset recorded time sent since last command
+function resetTimeSinceCMD() {
+  lastCMDSent = 0
+}
+
+// get and record time of last command sent
+function setTimeSinceCMD() {
+  lastCMDSent = new Date().getTime()
```

Comment:
given that this function is called a lot. Perhaps you should recycle the date object since its spamming the creation of this object. Maybe we don't care though, but it can help with performance. @PeterGhimself @vashmata

Reply:
Depends on whether you need a new Date object each time you check the time. I don't know how Date works

Reply:
I found a date object that is already preexisting on page load in modern browsers. So I just use that instead to get the current time.